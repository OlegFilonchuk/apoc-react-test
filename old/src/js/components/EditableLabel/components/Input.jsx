import React from 'react';
import PropTypes from 'prop-types';

import { KEY_CODES } from '../../../utils/constants';

function Input({ children, maxLength, readOnly, onSubmit, onCanceled, onChange, onFocus, alreadyFocused, value }) {
  const setFocus = node => {
    if (node && !alreadyFocused) {
      node.focus();
      node.select();
    }
  };

  const onKeyUp = e => {
    if (e.keyCode === KEY_CODES.ENTER) {
      onSubmit(e.target.value);
    } else if (e.keyCode === KEY_CODES.ESCAPE) {
      onCanceled(e.target.value);
    }
  };

  const onBlur = e => {
    onSubmit(e.target.value);
  };

  const onInputChange = e => {
    onChange(e.target.value);
  };

  const onInputFocus = e => {
    onFocus(e.target.value);
  };

  return (
    <input
      ref={setFocus}
      type="text"
      defaultValue={children}
      onBlur={onBlur}
      onKeyUp={onKeyUp}
      readOnly={readOnly}
      maxLength={maxLength}
      onChange={onInputChange}
      onFocus={onInputFocus}
      value={value}
    />
  );
}

Input.propTypes = {
  /**
   * Label text
   */
  children: PropTypes.string.isRequired,

  /**
   * Maximum allowed characters for input value
   */
  maxLength: PropTypes.number,

  /**
   * If input should be blocked from editing
   */
  readOnly: PropTypes.bool,

  /**
   * Fired when new value is submitted (via enter key or blur). Good place to run some validation.
   * It will pass current input value.
   */
  onSubmit: PropTypes.func.isRequired,

  /**
   * Fired when user cancels editing.
   */
  onCanceled: PropTypes.func.isRequired,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
  alreadyFocused: PropTypes.bool,
  value: PropTypes.string.isRequired
};

Input.defaultProps = {
  maxLength: 0,
  readOnly: false,
  onChange: () => true,
  onFocus: () => true,
  alreadyFocused: false
};

export default Input;
