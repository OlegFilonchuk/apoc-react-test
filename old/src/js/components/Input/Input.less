@import '../../utils/mixins/formElements';
@import '~chameleon/less/variables.less';

@counter-min-width: 30px;
@text-counter-distance: 6px;

@hidden: #b2b2b2;

.input-field {
  .form-element;

  &-wrapper {
    display: inline-block;
    position: relative;
  }

  &__password-eye {
    position: absolute;
    display: inline-block;

    right: 0.5em;
    top: 50%;
    transform: translate(0, -50%);

    font-size: 18px;
    cursor: pointer;
    &:hover {
      color: @primary-hover;
    }
    &__visible {
      color: @primary;
    }
    &__hidden {
      color: @hidden;
    }
  }

  &__char-limit-counter {
    position: absolute;
    right: 1px;
    top: 1px;
    min-width: @counter-min-width;
    height: calc(100% - 2px);
    background-color: #eaeaea;
    &__text {
      position: relative;
      top: 50%;
      margin-top: -30%;
      text-align: center;
    }
  }

  .input-suffix-wrapper {
    display: flex;
  }

  .input-field-suffix {
    margin-left: 10px;
    font-size: x-small;
    color: #a4a3a3;
  }
}

.with-characters-limit-input {
  padding-right: calc(@counter-min-width + @text-counter-distance);
}

input[type='password'],
input[type='text'] {
  &.extra-password-padding {
    padding-right: 32px;
  }
}
