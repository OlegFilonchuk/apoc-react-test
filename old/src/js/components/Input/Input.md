# Basic example
    (
      <Input placeholder="Please provide name" />
    )

# Password field example
    (
      <Input placeholder="Please provide name" type="password" />
    )
# Input with error
    (
      <Input hasError placeholder="Provide valid name" />
    )
# Uncontrolled component example
If you want to use Input as [uncontrolled component](https://facebook.github.io/react/docs/uncontrolled-components.html) you can provide `defaultValue` prop to set the initial value

    (
      <Input defaultValue="Some default value" />
    )

# Controlled component example
If you want to use Input as [Controlled component](https://facebook.github.io/react/docs/forms.html) you have to handle logic by yourself

    const initialState = {
      value: 'Initial Value'
    };

    const onChange = (event) => {
      const value = event.target.value;

      setState({ value });
      console.log('I have been change to: ', value);
    };

    (
      <Input value={state.value} onChange={onChange} />
    )
