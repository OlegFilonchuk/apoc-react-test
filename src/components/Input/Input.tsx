import React from 'react';

export type InputProps = {
  placeholder: string;
};

export function Input(props: InputProps) {
  return <input placeholder={props.placeholder} />;
}
