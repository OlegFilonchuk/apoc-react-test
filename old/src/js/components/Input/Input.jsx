import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { KEY_CODES } from '../../utils/constants';

import './Input.less';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

const TYPE_PASSWORD = 'password';

class Input extends Component {
  constructor(props) {
    super(props);

    this.state = {
      contentVisible: false
    };
  }

  onEyeClick = () => this.setState({ contentVisible: !this.state.contentVisible });

  get passwordIconClassNames() {
    return classNames('input-field__password-eye sc-icon-pw-show-ctr', {
      'input-field__password-eye__visible': this.state.contentVisible,
      'input-field__password-eye__hidden': !this.state.contentVisible
    });
  }

  render() {
    const { className, hasError, type, onInputRef, lettersLeftCount, submitOnEnter, suffix, ...rest } = this.props;

    const extendedClassName = classNames('sc-form-ele input-field', className, { error: hasError });
    let currentInputType = type;

    if (type === TYPE_PASSWORD && this.state.contentVisible) {
      currentInputType = 'text';
    }

    const isWithCharactersLimit = lettersLeftCount !== Infinity;

    const isWithSuffix = suffix !== '';

    const baseInputClassNames = classNames(extendedClassName, {
      'with-characters-limit-input': isWithCharactersLimit,
      'extra-password-padding': this.props.type === 'password'
    });

    const disableEnterSubmit = e => {
      if (!submitOnEnter && e.keyCode === KEY_CODES.ENTER) {
        e.preventDefault();
      } else {
        this.props.onKeyDown(e);
      }
    };

    const propsObj = {
      ...rest,
      className: baseInputClassNames,
      type: currentInputType,
      ref: onInputRef,
      ...getDataTestElementProps(this.props),
      onKeyDown: disableEnterSubmit
    };
    const baseInput = <input {...propsObj} />;

    if (type === TYPE_PASSWORD) {
      return (
        <span className="input-field-wrapper">
          {baseInput}
          <span className={this.passwordIconClassNames} onClick={this.onEyeClick} />
        </span>
      );
    }

    if (isWithCharactersLimit) {
      return (
        <span className="input-field-wrapper">
          {baseInput}
          <div className="input-field__char-limit-counter">
            <div className="input-field__char-limit-counter__text">{lettersLeftCount}</div>
          </div>
        </span>
      );
    }

    if (isWithSuffix) {
      return (
        <span className="input-suffix-wrapper">
          {baseInput}
          <div className="input-field-suffix">
            <div className="">{suffix}</div>
          </div>
        </span>
      );
    }

    return baseInput;
  }
}

Input.propTypes = {
  /**
   * If Input should render in error mode
   */
  hasError: PropTypes.bool,

  /**
   * Any custom class names
   */
  className: PropTypes.string,

  /**
   * Input type
   */
  type: PropTypes.string,

  /**
   * Callback to run with raw input reference as argument
   */
  onInputRef: PropTypes.func,

  /**
   * If is not default value (Infinity), then Input should also display letters limit
   *
   */
  lettersLeftCount: PropTypes.number,
  onKeyDown: PropTypes.func,
  submitOnEnter: PropTypes.bool,

  /**
   * Suffix string
   */
  suffix: PropTypes.string,
  ...dataTestElementPropTypes
};

Input.defaultProps = {
  hasError: false,
  className: '',
  type: 'text',
  onInputRef: element => element,
  lettersLeftCount: Infinity,
  onKeyDown: () => true,
  submitOnEnter: false,
  suffix: ''
};

export default Input;
