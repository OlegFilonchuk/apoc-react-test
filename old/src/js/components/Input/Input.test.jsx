import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Input from './Input';

describe('<Input />', () => {
  const renderInput = (props = {}, mountFn = shallow) => mountFn(<Input {...props} />);

  describe('snapshot tests', () => {
    it('should render default Input', () => {
      const input = renderInput();

      expect(toJson(input)).toMatchSnapshot();
    });

    it('should render Input with custom className', () => {
      const input = renderInput({ className: 'Some class name' });

      expect(toJson(input)).toMatchSnapshot();
    });

    it('should render Input with error', () => {
      const input = renderInput({ hasError: true });

      expect(toJson(input)).toMatchSnapshot();
    });
  });
});
