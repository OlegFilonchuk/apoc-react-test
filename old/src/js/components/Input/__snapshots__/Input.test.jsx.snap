// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> snapshot tests should render Input with custom className 1`] = `
<input
  className="sc-form-ele input-field Some class name"
  onKeyDown={[Function]}
  type="text"
/>
`;

exports[`<Input /> snapshot tests should render Input with error 1`] = `
<input
  className="sc-form-ele input-field error"
  onKeyDown={[Function]}
  type="text"
/>
`;

exports[`<Input /> snapshot tests should render default Input 1`] = `
<input
  className="sc-form-ele input-field"
  onKeyDown={[Function]}
  type="text"
/>
`;
